### ISSUE TITLE
0.6.9 fails to compile with GCC 6.4.0, 0.6.8 still worked

### ISSUE THREAD

**Issue by rse** [Tyriar/node-pty#108]
version 0.6.9 fails to compile with at least GCC 6.4.0 under FreeBSD, 0.6.8 still worked:

```
gmake: Entering directory '/u/rse/prj/node/blessed-xterm/node_modules/node-pty/build'
  CXX(target) Release/obj.target/pty/src/unix/pty.o
../src/unix/pty.cc: In function 'Nan::NAN_METHOD_RETURN_TYPE PtyFork(Nan::NAN_METHOD_ARGS_TYPE)':
../src/unix/pty.cc:206:22: error: 'UTF8' was not declared in this scope
     term->c_iflag |= UTF8;
                      ^~~~
../src/unix/pty.cc:206:22: note: suggested alternatives:
In file included from ../../nan/nan.h:47:0,
                 from ../src/unix/pty.cc:20:
/u/rse/.node-gyp/8.1.4/include/node/node.h:301:23: note:   'UTF8'
 enum encoding {ASCII, UTF8, BASE64, UCS2, BINARY, HEX, BUFFER, LATIN1 = BINARY};
                       ^~~~
In file included from ../src/unix/pty.cc:20:0:
../../nan/nan.h:1736:23: note:   'UTF8'
 enum Encoding {ASCII, UTF8, BASE64, UCS2, BINARY, HEX, BUFFER};
                       ^~~~
```

**Comment by Tyriar** [comment]
I'm the only one that changed that file in v0.6.9 in https://github.com/Tyriar/node-pty/commit/abb0a37c67228cd4a2ac669de657c295e6c4be2c#diff-32a8618c58cb849e9b5535ba19ffacc5, but i wouldn't expect it to be using the `UTF8` from node though http://man7.org/linux/man-pages/man3/termios.3.html

@rse could you check on FreeBSD if changing that line to `v8::UTF8` or `node::UTF8` fixes it?

@KiNgMaR perhaps this is an obscure bug that we're using the `node::UTF8` enum? Is this what you were expecting in https://github.com/Tyriar/node-pty/commit/57f4866276b80bb7785a87d703aa162cebadf518?

**Comment by rse** [comment]
Well, your UTF8 should be a terminal mode. You can't use the Node UTF8 or V8 UTF8 enums. Their number stands for a different thing. The point simply is: in 0.6.8 there was no IUTF8/UTF8 usage at all. Now in 0.6.9 there is:

```
#if defined(IUTF8)
    term->c_iflag |= IUTF8;
#else
    term->c_iflag |= UTF8;
#endif
```

This indicated that if no IUTF8 is defined, then UTF8 is. It should be at least:

```
#if defined(IUTF8)
    term->c_iflag |= IUTF8;
#elif defined(UTF8)
    term->c_iflag |= UTF8;
#endif
```

But I think this is also wrong, because "UTF8" can be an arbitrary definition not related to terminal modes. So, IMHO the correct thing is:

```
#if defined(IUTF8)
    term->c_iflag |= IUTF8;
#endif
```

In other words: If no IUTF8 exists on a Unix system, you cannot do anything except to ignore this fact...

**Comment by Tyriar** [comment]
@rse yeah, I think my removing of the `using namespace` has had exactly the effect that I wanted; it revealed that line is either doing nothing or potentially doing harm as we're using the number from `node::UTF8`.

If you believe your last example is the most correct then let's go with that, you seem to know more about this than me tbh :smile: Would you like to submit a PR or want me to make the change?

**Comment by rse** [comment]
Yes, please be so kind and change...

```
#if defined(IUTF8)
    term->c_iflag |= IUTF8;
#else
    term->c_iflag |= UTF8;
#endif
```
...to just:

```
#if defined(IUTF8)
    term->c_iflag |= IUTF8;
#endif
```

Then please release a new version 0.9.10 and this way node-pty is no longer broken on all Unix systems without IUTF8. Thanks!

**Comment by Tyriar** [comment]
@rse published 0.6.10, please let me know if there's an issue :smile:

**Comment by flakes** [comment]
Guess I'm a bit late to the party, but no, that's not what I was expecting in 57f4866276b80bb7785a87d703aa162cebadf518. I simply moved the common part outside of the #ifdef there :)

**Comment by rse** [comment]
0.9.10 works just fine out-of-the-box again. Thanks for the fix.

**Comment by Tyriar** [comment]
Thanks for reporting quickly!
